File: C#/BA1C.cs
using System;

namespace BA1C
{
    class Program
    {

    //A solution to a ROSALIND bioinformatics problem.
    //Problem Title: Find the reverse complement of a DNA string
    //Rosalind ID: BA1C
    //URL: http://rosalind.info/problems/ba1c/

        static void Main(string[] args)
        {
            string x = "AAAACCCGGT";
            Console.WriteLine(reverse(complement(x)));

            string complement(string text)
            {
                string compl = "";
                int nt = text.Length;
                for (int i = 0; i < nt; i++)
                {
                    if (text[i] == 'G')
                    {
                        compl += "C";
                    }
                    if (text[i] == 'C')
                    {
                        compl += "G";
                    }
                    if (text[i] == 'A')
                    {
                        compl += "T";
                    }
                    if (text[i] == 'T')
                    {
                        compl += "A";
                    }
                }
                return compl;
            }
            string reverse(string text)
            {
                string rev = "";
                int nt = text.Length;
                for (int i = nt - 1; i > -1; i--)
                {
                    rev = rev + text[i];
                }
                return rev;
            }

        }
    }
}

File: Python/BA1C.py
"""
A solution to a ROSALIND bioinformatics problem.
Problem Title: Find the reverse complement of a DNA string
Rosalind ID: BA1C
URL: http://rosalind.info/problems/ba1c/
"""

def complement(text):
    """complement of text"""
    compl=""
    nt = len(text)
    for i in range (0,nt):
        if text[i]=="G":
            compl=compl+"C"
        if text[i]=="C":
            compl=compl+"G"
        if text[i]=="A":
            compl=compl+"T"
        if text[i]=="T":
            compl=compl+"A"
    return compl

def reverse(text):
    """reverse of text"""
    return text[::-1]

if __name__ == '__main__':
    x="AAAACCCGGT"
    print (str(reverse(complement(x))))
